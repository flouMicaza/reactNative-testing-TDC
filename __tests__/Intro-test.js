// __tests__/Intro-test.js
import React from 'react';
import Intro from '../Intro';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {StyleSheet, Text, View} from 'react-native';
//
Enzyme.configure({ adapter: new Adapter() });
// jest.mock('react-navigation/src/routers/KeyGenerator', () => ({
//   generateKey: jest.fn(() => 123),
// }));

describe('test-name',()=>{

  test('find text',()=>{
    const wrapper = shallow(<Intro/>);
    expect(wrapper.find(Text).at(0).props().children).toBe("Welcome to React Native!");
  })


  test('find text2',()=>{
      const wrapper = shallow(<Intro/>);
      expect(wrapper.find(Text).at(1).props().children).toBe("This is a React Native snapshot test.");
    })


  test('use shallow correctly',()=>{
    const wrapper = shallow(<Intro/>);
    expect(wrapper).toMatchSnapshot();
  })

  test('renders correctly', () => {
    const tree = renderer.create(<Intro />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('adding positive numbers is not zero', () => {
    for (let a = 1; a < 10; a++) {
      for (let b = 1; b < 10; b++) {
        expect(a + b).not.toBe(0);
      }
    }
  });

  test('there is no I in team', () => {
    expect('team').not.toMatch(/I/);
  });

  test('but there is a "stop" in Christoph', () => {
    expect('Christoph').toMatch(/stop/);
  });
})
