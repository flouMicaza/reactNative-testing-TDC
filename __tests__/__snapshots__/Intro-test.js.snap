// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-name renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="text-thing"
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test.
  </Text>
</View>
`;

exports[`test-name use shallow correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Intro />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Welcome to React Native!
        </Component>,
        <Component
          style={
            Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          This is a React Native snapshot test.
        </Component>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      },
      "testID": "text-thing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Welcome to React Native!",
          "style": Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Welcome to React Native!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "This is a React Native snapshot test.",
          "style": Object {
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "This is a React Native snapshot test.",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Welcome to React Native!
          </Component>,
          <Component
            style={
              Object {
                "color": "#333333",
                "marginBottom": 5,
                "textAlign": "center",
              }
            }
          >
            This is a React Native snapshot test.
          </Component>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
        },
        "testID": "text-thing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Welcome to React Native!",
            "style": Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Welcome to React Native!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "This is a React Native snapshot test.",
            "style": Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "This is a React Native snapshot test.",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
